pkg/plugin/shared_health_check.go

```diff
@@ -0,0 +1,76 @@
+package plugin
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// TODO: make configurable
+const (
+	DefaultHealthCheckPeriod = 30 * time.Second
+	DefaultErrcBufSize       = 100
+)
+
+type SharedHealthCheck struct {
```

Comment:
nit: Why is this called `Shared`?

Reply:
discussed offline, the health check is shared across v1&v2